website/blog/2023-09-05-deploy-tabby-to-huggingface-space/index.md

```diff
@@ -0,0 +1,125 @@
+**Everything will be done in the Hugging Face UI. No local setup is needed.**
+
+### Deep Dive
+
+#### Create a new Space
+
+After you create a Hugging Face account, you should be able to see the following page by clicking this [link](https://huggingface.co/new-space).
+The owner name will be your account name. Fill in a Space name, e.g. "tabbyml", and select Docker as Space SDK. Then click "Create Space" at the bottom.
+
+![Create a new Space](./new-space.png)
+
+#### Uploading Dockerfile
+
+For advanced users, you can leverage the Git workspace. In this blog post, we will show you the UI flow instead. After you click the "Create a Space" in the last step, you will be directed to this page. Just ignore the main text and click the "Files" on the top right corner.
+
+![Docker Space](./docker-space.png)
+
+After clicking on the "Files", you will be able to see a "Add file" button, click that, then click on "Create a new file"
+
+![Empty Space](./empty-space.png)
+
+Then you will be redirected to the page below. Set the filename to “Dockerfile” and copy the content to the “Edit” input box. You can copy the code from the [appendix](#dockerfile) here. Once ready, click the button “Commit new file to main” on the bottom.
+
+![Edit Dockerfile](./edit-dockerfile.png)
+
+#### Edit Readme
+
+You also need to add a new line the README.md. Click the “edit” button in the README.md file.
+
+
+![Empty README](./empty-readme.png)
+
+Add this line “app_port: 8080” after “sdk: docker”
+
+![Edit README](./edit-readme.png)
+
+Click the button “Commit to main” to save the changes.
+
+#### Verify Tabby is running
+
+Click on the “App” button, you should be able to see the container is building:
+
+![Space Building](./building.png)
+
+If the App is up successfully, you should see this page:
+
+![Tabby Swagger](./tabby-swagger.png)
+
+#### Call the API to do Code Completion
+
+Now, you are able to call the completion API. The full URL is https://{YOUR-ACCOUNT-NAME}-tabbyml.hf.space/v1/completions. In this post, the URL is https://randxie-tabbyml.hf.space/v1/completions.
+
+To test if your APIs are up and running, use [this online tool](https://reqbin.com/curl) to send curl commands:
+
+![curl](./curl.png)
+
+The full curl command can be found in the [appendix](#curl-command) as well. Make sure you have changed the URL to match with your Hugging Face Spaces settings!!
+
+### Conclusion
+In this post, we covered the detailed steps for deploying a Tabby instance to Hugging Face Spaces. By following these steps, anyone is able to bring up their own code completion APIs easily.
+
+### Appendix
+
+#### Dockerfile
+```Dockerfile
+FROM tabbyml/tabby
+
+RUN useradd -m -u 1000 user
+
+USER user
+
+ENV HOME=/home/user \
+    PATH=/home/user/.local/bin:$PATH
+
+ENV TABBY_ROOT=$HOME/app
+
+WORKDIR $HOME/app
+
+COPY --chown=user . $HOME/app
+
+CMD ["serve", "--model", "TabbyML/SantaCoder-1B"]
```

Comment:
This should be:

```
"--device", "cuda", "--model", "TabbyML/CodeLlama-7B", "--compute-type", "int8"
```

I tried with the CPU version. The app can be brought up but very slow. Using GPU as default is a good idea given that it's a 7B model.